### PR TITLE
hangman game

### PR DESCRIPTION
**Pull request by tdc2090**
Very simplified version of hangman. Not allowed to use Arrays, ArrayLists or StringBuilder.

### CODE REVIEW

File: src/Hangman.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
+import java.util.Scanner;
+
+/**
```
Comment (shaunnetherby) [inline src/Hangman.java:3]: Remove comment

File: src/Hangman.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            System.out.println("\nThe disguised word is <" + getDisguisedWord() + ">");
+            makeGuess(promptLetterGuess());
+            System.out.println("Guess made " + numberOfGuesses + " with " + numberOfIncorrectGuesses + " wrong (" + incorrectGuesses + ")");
+
+        }
+
+        System.out.println("Congratulations, you found the secret word: " + getSecretWord());
+    }
+
+    public static void main(String[] args) { Hangman game = new Hangman(); game.initialize("Happiness");
```
Comment (paoloaprile) [inline src/Hangman.java]: Whoa. Newline this please. My eyes were freaking out when I saw this.

File: src/Hangman.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        incorrectGuesses = "";
+        numberOfGuesses = 0;
+        numberOfIncorrectGuesses = 0;
+    }
+
+    private void makeGuess(char c) {
+        if (secretWord.contains("" + c)) {
+            disguisedWord = reformatDisguisedWord(c);
+            numberOfGuesses++;
+        }else{
```
Comment (shaunnetherby) [inline src/Hangman.java:64]: spacing

File: src/Hangman.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    private boolean isFound() {
+        return disguisedWord.equals(secretWord);
+    }
+
+    public void initialize(String word){
+        secretWord = word.toLowerCase();
+
+        for (int i = 0; i <= secretWord.length() - 1; i++) {
```
Comment (shaunnetherby) [inline src/Hangman.java:51]: `i < secretWord.length(); i++) {`

File: src/Hangman.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            numberOfGuesses++;
+        }else{
+            numberOfGuesses++;
+            numberOfIncorrectGuesses++;
+            incorrectGuesses += c;
+        }
+
+    }
+
+    private String reformatDisguisedWord(char c1) {
```
Comment (shaunnetherby) [inline src/Hangman.java:72]: `private String revealLetter(char letter) {`

File: src/Hangman.java [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+    private char promptLetterGuess() {
+        Scanner keyboard = new Scanner(System.in);
+        String guessedLetter;
+
+        System.out.print("Guess a letter: ");
+        guessedLetter = keyboard.next().toLowerCase();
+
+        if (disguisedWord.contains(guessedLetter) || incorrectGuesses.contains(guessedLetter)){
+            while (disguisedWord.contains(guessedLetter) || incorrectGuesses.contains(guessedLetter)) {
```
Comment (shaunnetherby) [inline src/Hangman.java:92]: I would write this in a way that you didn't have another while loop here.You could move some of the logic down into play game so `promptLetterGuess` does less

File: src/Hangman.java [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+                System.out.println("Already guessed that one, try again.");
+                System.out.print("Guess a letter: ");
+                guessedLetter = keyboard.next().toLowerCase();
+                numberOfGuesses++;
+            }
+        }
+        if (("" + guessedLetter).length() > 1){
```
Comment (shaunnetherby) [inline src/Hangman.java:99]: if (!guessedLetter.isEmpty())

**Pull request closed without merging**